cmd/tailscale/cli/funnel.go

```diff
@@ -73,6 +73,18 @@ func (e *serveEnv) runFunnel(ctx context.Context, args []string) error {
 	default:
 		return flag.ErrHelp
 	}
+
+	prefs, err := e.lc.GetPrefs(ctx)
+	if err != nil {
+		return fmt.Errorf("getting prefs: %w", err)
+	}
+
+	if on && prefs.ShieldsUp {
+		return fmt.Errorf("%s\n%s",
+			"Unable to turn on Funnel while shields-up=true.",
+			"Run `tailscale set --shields-up=false` to turn off Shields.")
+	}
```

Comment:
I think you want to do this in local.go
https://github.com/tailscale/tailscale/blob/f4da995940ea13bda69979ec1a07251c6b757c5d/ipn/ipnlocal/local.go#L2532